.circleci/config.yml

```diff
@@ -208,9 +208,44 @@ commands:
 
   run_build:
     steps:
+      # TODO: remove the line below
+      # Before merging this PR we need to use an alternative develop branch to test the functionality, `develop-cache-test`
+
+      # On `develop-cache-test` we intentionally do NOT restore any ccache, so every run starts from scratch.
+      # Other branches restore the most recent cache produced on `develop-cache-test`.
+      - when:
+          condition:
+            not:
+              equal: ["develop-cache-test", << pipeline.git.branch >>]
+          steps:
+            - restore_cache:
+                keys:
+                  # Reuse cache produced on develop-cache-test (prefix match; restores the most recent).
+                  - v1-ccache-{{ arch }}-develop-cache-test-
+                  - v1-ccache-{{ arch }}-
       - run:
           name: Build
-          command: scripts/ci/build.sh
+          command: |
+            export CCACHE_DIR="$HOME/.ccache"
+            export CCACHE_BASEDIR="$(pwd)"
+            export CCACHE_NOHASHDIR=1
+            mkdir -p "$CCACHE_DIR"
+            export CMAKE_OPTIONS="${CMAKE_OPTIONS:-} -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache"
+            ccache -z || true
```

Comment:
Do we expect not to have this on any of the platforms?
Is that platform Windows? :P